File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        mx=0
        minp=float('inf')
        for p in prices:
            if p<minp:
                minp=p
            elif p-minp >mx:
                mx=p-minp
        return mx
        

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int max=0;
        int min=Integer.MAX_VALUE;
        for(int i=0;i<prices.length;i++){
            if(prices[i]<min){
                min=prices[i];
            }
            int prof=prices[i]-min;
            if(prof>max){
                max=prof;
            }
        }
        return max;
    }
}